kotlinx-coroutines-core/jvm/src/scheduling/CoroutineScheduler.kt

```diff
@@ -78,23 +86,26 @@ internal class CoroutineScheduler(
         }
     }
 
-    private val globalQueue: GlobalQueue = GlobalQueue()
+    @JvmField
+    val globalCpuQueue = GlobalQueue()
+    @JvmField
+    val globalBlockingQueue = GlobalQueue()
```

Comment:
When I am going to incorporate long scanning, we could get rid of global queues and replace it with sparse `WorkerQueue` at uneven indices in `workers` with CAS-tryLock protection and push external tasks to these queues

Reply:
Nah, apparently it can be done properly only with access to `threadLocalRandomProbe`